### ISSUE TITLE
Change tooltip instantaneously

### ISSUE THREAD

**Issue by MrGlasspoole** [thednp/bootstrap.native#300]
How can i change the tooltip without the need to un-hover and hover again?

This changes the tooltip but doesn't show it instantaneously:
```
<button id="copyIt" type="button" data-clipboard-action="copy" data-clipboard-target="#toCopy" data-toggle="tooltip" data-placement="top" title="Link kopieren">Copy</button>

const copyIt = document.getElementById('copyIt');

clipboard.on('success', function(e) {
  linkCopied();
});

function linkCopied() {
  copyIt.setAttribute('data-original-title', 'Der Link wurde kopiert!');
};
```

**Comment by thednp** [comment]
You can make use of the `show` event to change the `data-original-title` and the content of the tooltip itself.

**Comment by MrGlasspoole** [comment]
I must be missing something.
I get "ReferenceError: tooltip is not defined"

**Comment by thednp** [comment]
I suggest you have a look at the documentation and fill in the blanks.

**Comment by MrGlasspoole** [comment]
What does "fill in the blanks" mean?

Before i ask and bother somebody i try stuff myself for hours and i did read the documentation 50 times.

tooltipInit gives me the same error.

In original Bootstrap all there is to do is:
```.attr('data-original-title', 'Der Link wurde kopiert!').tooltip('show');```

I tried many different combinations like:
```
var myTooltip = document.querySelectorAll('[data-toggle="tooltip"]');
var tooltipInit = myTooltip.Tooltip;

clipboard.on('success', function(e) {
  tooltipInit.hide()
  copyIt.setAttribute('data-original-title', 'Der Link wurde kopiert!');
  tooltipInit.show()
});
```

```
var elementsTooltip = document.querySelectorAll('[data-toggle="tooltip"]');

for (var i = 0; i < elementsTooltip.length; i++){
  new Tooltip(elementsTooltip[i])
}

clipboard.on('success', function(e) {
  tooltip.hide()
  copyIt.setAttribute('data-original-title', 'Der Link wurde kopiert!');
  tooltip.show()
});
```

**Comment by thednp** [comment]
Your code is a perfect word by word jQuery to vanilla, which is wrong.

```
var myTooltip = document.querySelectorAll('[data-toggle="tooltip"]');
var tooltipInit = myTooltip.Tooltip; // THIS is gonna throw error
```

**Comment by MrGlasspoole** [comment]
I don't get it. That line is from the docs.

**Comment by thednp** [comment]
This is from documentation

```js
// find all elements with title attribute THIS IS A COLLECTION
var elementsTooltip = document.querySelectorAll('[title]');
// var elementsTooltip = document.querySelectorAll('[data-toggle="tooltip"]'); // alternative

// attach a tooltip for each
for (var i = 0; i < elementsTooltip.length; i++){
  new Tooltip(elementsTooltip[i], {
    placement: 'top', //string
    animation: 'slideNfade', // CSS class
    delay: 150, // integer
  })
}
```

For handling a specific tooltip target

```js
// find an element initialized with Tooltip
var myLinkWithTooltip = document.getElementById('myLinkWithTooltip');

// reference the initialization object
var myTooltipInit = myLinkWithTooltip.Tooltip;
```

See the difference?

**Comment by MrGlasspoole** [comment]
There is something really wrong.
I made a simple test with just:
```
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title>Untitled Document</title>

<link rel="stylesheet" type="text/css" href="bootstrap.min.css" />
<style>
.btn {
  display:block;
  margin-top:200px;
  margin-left:auto;
  margin-right:auto;
}
</style>
</head>

<body>

<button class="btn" type="button" data-toggle="tooltip" title="Link kopieren">Copy</button>

<script src="native.min.js"></script>
<script>
var elementsTooltip = document.querySelectorAll('[data-toggle="tooltip"]');

for (var i = 0; i < elementsTooltip.length; i++){
  new Tooltip(elementsTooltip[i], {
    placement: 'bottom', //string
    animation: 'slideNfade', // CSS class
    delay: 150, // integer
  })
}
</script>
</body>
</html>
```
No matter what i put in placement, the tip is always on top...

**Comment by thednp** [comment]
If the tooltip cannot be shown on top, it will be displayed where there is enough viewport room, regardless of the setting, but we were talking about a very different issue here.

**Comment by MrGlasspoole** [comment]
I did that for testing to make sure nothing else is in the way.

But even that simple thing does not work right.
I button in the middle of a page with 200px above and ~950px to the left and right should have enough room.

**Comment by thednp** [comment]
OK I read the post and understand, so here's what happens:

* using the `data-toggle` attribute and a `title` attribute for the button, the component will automatically init the element if you're using the build from the master
* then doin the re-init will sometimes mess up with the options for some reason I can't say out of my head right now
* in comparison with the original plugin, our component will init on page load all eligible component targets
* in conclusion you're better off not using the `data-toggle` attribute and pick this specific via a different selector and init with your options

Let me know how this goes.

**Comment by MrGlasspoole** [comment]
Ok, changing to "data-mytoggle="tooltip" makes the javascript work so placement and delay is working.
But still no show and hide:
```
var elementsTooltip = document.querySelectorAll('[data-mytoggle="tooltip"]');

for (var i = 0; i < elementsTooltip.length; i++){
  new Tooltip(elementsTooltip[i], {
    placement: 'left', //string
    animation: 'slideNfade', // CSS class
    delay: 150, // integer
  })
}

const clipboard = new ClipboardJS('#copyIt');
const copyIt = document.getElementById('copyIt');

clipboard.on('success', function(e) {
  myTooltipInit.hide()
  copyIt.setAttribute('data-original-title', 'Der Link wurde kopiert!');
  myTooltipInit.show()
});
```
This gives "tooltipInit is not defined".
To make sure I'm not missing anything i even tried `elementsTooltip.hide()` and get it is not a function.
I even tried stuff like `var myTooltipInit = Tooltip;` or `var myTooltipInit = elementsTooltip;`

Testing with the DATA API shows nor errors:
```
const copyIt = document.getElementById('copyIt');
var myTooltipInit = copyIt.Tooltip;

const clipboard = new ClipboardJS('#copyIt');

clipboard.on('success', function(e) {
  myTooltipInit.hide()
  copyIt.setAttribute('data-original-title', 'Der Link wurde kopiert!');
  myTooltipInit.show()
});
```
But the tooltip is not hidden and then shown. Only the attribute is set.

**Comment by thednp** [comment]
Why `querySelectorAll`? why not handle your individual case in a single object? I can now understand your vanilla JS skill is not very sharp yet.

Here my adjustments to your code

```js
var elementTooltip = document.querySelector('[data-mytoggle="tooltip"]');

// you didn't define myTooltipInit anywhere man
const myTooltipInit = new Tooltip(elementsTooltip[i], {
  placement: 'left', //string
  animation: 'slideNfade', // CSS class
  delay: 150, // integer
})

const clipboard = new ClipboardJS('#copyIt');
const copyIt = document.getElementById('copyIt');

clipboard.on('success', function(e) {
  myTooltipInit.hide()
  copyIt.setAttribute('data-original-title', 'Der Link wurde kopiert!');
  myTooltipInit.show()
});
```

But to make it work with multiple tooltip contents, you would work better with a fork that works your way, which by the way, Popover component should be able to do thanks to the template system.

**Comment by MrGlasspoole** [comment]
No errors here. But the tooltip is still not hidden and then shown.

At the moment for that single tooltip a single object is enough. So also via DATA API and getElementById would make me happy.

**Comment by thednp** [comment]
Doing the custom work for yourself would make me even more happy, and you as well in the long run.

**Comment by MrGlasspoole** [comment]
Custom work? So `myTooltipInit.hide()` and `myTooltipInit.show()` is not enough?
I thought that will do it? If not I'm missing a big part and no wonder it does not work after hours of trying.

**Comment by thednp** [comment]
If I were you I would spend more hours in learning.

**Comment by MrGlasspoole** [comment]
Learning by doing.
But i don't understand why `myTooltipInit.hide()` does not hide it.

**Comment by thednp** [comment]
Well if you would have made an example of your context on codepen or jsfidle, we would better understand and provide support, but that won't work either if you don't understand JavaScript.
